src/zenml/zen_server/routers/steps_endpoints.py

```diff
@@ -142,10 +142,16 @@ def get_step(
     Returns:
         The step.
     """
-    step = zen_store().get_run_step(step_id, hydrate=hydrate)
+    # We always fetch the step hydrated because we need the pipeline_run_id
+    # for the permission checks. If the user requested an unhydrated response,
+    # we later remove the metadata
+    step = zen_store().get_run_step(step_id, hydrate=True)
     pipeline_run = zen_store().get_run(step.pipeline_run_id)
```

Comment:
I know it is in the metadata, but ideally, shouldn't the `pipeline_run_id` already be in the body of the response?

Reply:
Ideally it would be, but then this might be breaking some stuff in the frontend which I don't want to risk right now